contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -52,6 +52,11 @@ const (
 	pluginRecoveryDelay = 100 * time.Millisecond // delay after scheduler plugin crashes, before we resume scheduling
 )
 
+const (
+	FailedScheduling = "FailedScheduling"
+	Scheduled        = "Scheduled"
+)
+
 // scheduler abstraction to allow for easier unit testing
 type schedulerInterface interface {
```

Comment:
I did this :(  I needed to stop the plugin from reaching directly for the
fields of KubernetesScheduler. I wasn't feeling very creative or inspired.
schedulerInterface was named. Ideally the plugin stuff (the k8s side of the
scheduler) would live in a plugin/ package. I use the name "plugin" because
the upstream scheduler is in a "plugin/scheduler/..." set of packages so it
was easy for me to distinguish between the "mesos" side of the brain and
the "plugin" side of the brain.

On Tue, Sep 22, 2015 at 11:40 AM, Karl Isenberg <notifications@github.com>
wrote:

> In contrib/mesos/pkg/scheduler/plugin.go
> <https://github.com/kubernetes/kubernetes/pull/14271#discussion_r40102215>
> :
>
> > @@ -52,6 +52,11 @@ const (
> >  	pluginRecoveryDelay = 100 * time.Millisecond // delay after scheduler plugin crashes, before we resume scheduling
> >  )
> >
> > +const (
> > +	FailedScheduling = "FailedScheduling"
> > +	Scheduled        = "Scheduled"
> > +)
> > +
> >  // scheduler abstraction to allow for easier unit testing
> >  type schedulerInterface interface {
>
> I'm really not a fan of this naming convention using -Interface.
> Are we doing this throughout k8sm? Is the rest of k8s doing this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/14271/files#r40102215>.
>